Source/Scene/LabelCollection.js

```diff
@@ -638,6 +713,11 @@ define([
         this.removeAll();
         this._billboardCollection = this._billboardCollection.destroy();
         this._textureAtlas = this._textureAtlas && this._textureAtlas.destroy();
+
+        if (defined(this._removeEventFunc)) {
```

Comment:
The collection is getting destroyed so there's no need to listen on the event any longer.